lib/application/widgets/theme/rich_text_field.dart

```diff
@@ -0,0 +1,297 @@
+///
+/// Ideally this wrapper would not be necessary but `quill` forces us to provide a set of attributes when building a
+/// [quill.QuillEditor] that are shared between this file editor instances.
+///
+/// It basically reduces boilerplate.
+class _ThemedEditor extends HookWidget {
+  const _ThemedEditor({
+    required this.controller,
+    required this.backgroundColor,
+    required this.placeholder,
+    this.showCursor = true,
+    this.readonly = false,
+  });
+
+  final quill.QuillController controller;
+  final Color backgroundColor;
+  final String placeholder;
+  final bool showCursor;
+  final bool readonly;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = useTheme();
+    final textTheme = Theme.of(context).textTheme;
+    const zeroTuple = Tuple2<double, double>(0, 0);
+
+    return quill.QuillEditor(
+      controller: controller,
+      scrollController: ScrollController(),
+      scrollable: true,
+      focusNode: FocusNode(),
+      autoFocus: true,
+      readOnly: readonly,
+      expands: false,
+      padding: EdgeInsets.zero,
+      showCursor: showCursor,
+      // TODO(ggirotto): Placeholder is crashing. This is a problem related to `FlutterQuill`.
+      // TODO(ggirotto): https://github.com/singerdmx/flutter-quill/issues/348
+      // placeholder: placeholder,
+      customStyles: quill.DefaultStyles(
+        paragraph: quill.DefaultTextBlockStyle(textTheme.bodyText2!, zeroTuple, zeroTuple, null),
+        placeHolder: quill.DefaultTextBlockStyle(
+          textTheme.bodyText1!.copyWith(color: theme.neutralSwatch.shade400),
+          zeroTuple,
+          zeroTuple,
+          null,
+        ),
+        code: quill.DefaultTextBlockStyle(
+          textTheme.bodyText1!,
+          zeroTuple,
+          zeroTuple,
+          BoxDecoration(color: backgroundColor),
+        ),
+      ),
+    );
+  }
+}
+
+enum _ToolbarAttribute { bold, italic, underline, code }
+
+/// A toolbar of actions to customize the editor content.
+class _RichTextFieldToolbar extends HookWidget {
+  _RichTextFieldToolbar(this.controller);
+
+  final quill.QuillController controller;
+
+  /// Maps [_ToolbarAttribute] to `quill`s native [quill.Attribute] class.
+  final Map<_ToolbarAttribute, quill.Attribute<dynamic>> _toolBarAttributes = {
+    _ToolbarAttribute.bold: quill.Attribute.bold,
+    _ToolbarAttribute.italic: quill.Attribute.italic,
+    _ToolbarAttribute.underline: quill.Attribute.underline,
+    _ToolbarAttribute.code: quill.Attribute.codeBlock,
+  };
+
+  /// Maps [_ToolbarAttribute] to its respective asset.
+  final Map<_ToolbarAttribute, String> _toolBarAsset = {
+    _ToolbarAttribute.bold: images.bold,
+    _ToolbarAttribute.italic: images.italic,
+    _ToolbarAttribute.underline: images.underline,
+    _ToolbarAttribute.code: images.code,
+  };
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = useTheme();
+
+    final selectedAttributes = useState<List<_ToolbarAttribute>>([]);
```

Comment:
Ideally should be a Set. If order is required, you should guarantee unique items in the list or provide a way to sort the Set